apple/testing/default_runner/ios_xctestrun_runner.template.sh

```diff
@@ -384,32 +188,29 @@ fi
 if [[ "$should_use_xcodebuild" == true ]]; then
   if [[ -z "$test_host_path" && "$intel_simulator_hack" == true ]]; then
     echo "error: running x86_64 tests on arm64 macs using 'xcodebuild' requires a test host" >&2
+    echo "error: '$test_file'" >&2
     exit 1
   fi
 
-  readonly xctestrun_file="$test_tmp_dir/tests.xctestrun"
-  /usr/bin/sed \
-    -e "s@BAZEL_INSERT_LIBRARIES@$xctestrun_libraries@g" \
-    -e "s@BAZEL_TEST_BUNDLE_PATH@$xcrun_test_bundle_path@g" \
-    -e "s@BAZEL_TEST_ENVIRONMENT@$xctestrun_env@g" \
-    -e "s@BAZEL_TEST_HOST_BASED@$xctestrun_test_host_based@g" \
-    -e "s@BAZEL_TEST_HOST_PATH@$xctestrun_test_host_path@g" \
-    -e "s@BAZEL_TEST_HOST_BUNDLE_IDENTIFIER@$xcrun_test_host_bundle_identifier@g" \
-    -e "s@BAZEL_TEST_PRODUCT_MODULE_NAME@${test_bundle_name//-/_}@g" \
-    -e "s@BAZEL_IS_XCTRUNNER_HOSTED_BUNDLE@$xcrun_is_xctrunner_hosted_bundle@g" \
-    -e "s@BAZEL_IS_UI_TEST_BUNDLE@$xcrun_is_ui_test_bundle@g" \
-    -e "s@BAZEL_TARGET_APP_PATH@$xcrun_target_app_path@g" \
-    -e "s@BAZEL_TEST_ORDER_STRING@%(test_order)s@g" \
-    -e "s@BAZEL_DYLD_LIBRARY_PATH@__PLATFORMS__/$test_execution_platform/Developer/usr/lib@g" \
-    -e "s@BAZEL_COVERAGE_OUTPUT_DIR@$test_tmp_dir@g" \
-    -e "s@BAZEL_COMMAND_LINE_ARGS_SECTION@$xctestrun_cmd_line_args_section@g" \
-    -e "s@BAZEL_SKIP_TEST_SECTION@$xctestrun_skip_test_section@g" \
-    -e "s@BAZEL_ONLY_TEST_SECTION@$xctestrun_only_test_section@g" \
-    -e "s@BAZEL_ARCHITECTURE@$architecture@g" \
-    -e "s@BAZEL_TEST_BUNDLE_NAME@$test_bundle_name.xctest@g" \
-    -e "s@BAZEL_PRODUCT_PATH@$xcrun_test_bundle_path@g" \
-    "%(xctestrun_template)s" > "$xctestrun_file"
+  xctestrun_command_line_args=""
+  if [[ -n "${command_line_args:-}" ]]; then
+    xctestrun_command_line_args="${command_line_args[@]}"
+  fi
 
+  readonly xctestrun_file="$test_tmp_dir/tests.xctestrun"
+  test_env="$test_env" \
+    TEST_FILTER="$TEST_FILTER" \
+    command_line_args="$xctestrun_command_line_args" \
+    ./%(xctestrun_creator.sh)s \
```

Comment:
I don't love this arg and env var passing mechanism all that much. Let me know if you've better ideas. I figure we can make it more explicit in follow up work as we add usages of the xctestrun creation script.